msteams-platform/messaging-extensions/build-api-based-message-extension.md

```diff
@@ -11,399 +11,921 @@ ms.date: 10/19/2023
 
-1. Select any of the following options:
-   * If you want to start without an API, select **Start with a new API**.
-   * If you have an existing OpenAPI Description document, select **Start with an OpenAPI Description**.
+       :::image type="content" source="../assets/images/authentication/teams-sso-tabs/update-app-auth-msg.png" alt-text="Client application added message":::
 
-1. Select **Next**.
+       The authorized app's client ID displays on the page.
 
-   :::image type="content" source="../assets/images/Copilot/api-based-me-vs-create-project.png" alt-text="Screenshot shows the Search results from API, New API, OpenAPI Description Document, and Create options in Visual Studio to create a new Project.":::
+       :::image type="content" source="../assets/images/authentication/teams-sso-tabs/client-app-added.png" alt-text="Client app added and displayed":::
 
-1. Based on the options selected in **step 7**, select the following:
+> [!NOTE]
+> You can authorize more than one client application. Repeat the steps of this procedure for configuring another authorized client application.
+
+You've successfully configured app scope, permissions, and client applications. Ensure that you note and save the application ID URI. Next, you configure the access token version.
 
-   # [New API](#tab/new-api2)
+### Update app manifest
+
+> [!NOTE]
+> `webApplicationInfo` is supported inthe app manifest version 1.5 or later.
 
-   1. To get started, you must update the source code in the following files:
+Update the following properties in the app manifest file:
 
-      |File  |Contents |
-      |---------|---------|
-      |`repair.cs` | The main file of a function in Azure Functions. Defines an Azure Function that retrieves and filters repair records based on a query parameter from an HTTP GET request, and returns the results as a JSON response.|
-      |`RepairData.cs`|The data source for the repair API. Contains a method that returns a hardcoded list of car repair tasks. |
-      |`Models/RepairModel.cs`|Defines a data model that represents a repair task with properties such as ID, Title, Description, AssignedTo, Date, and Image.|
-      |`appPackage/apiSpecificationFiles/repair.yml` |  A file that describes the structure and behavior of the repair API.|
-      |`appPackage/responseTemplates/repair.json` |  A generated Adaptive Card that used to render API response.|
+* `webApplicationInfo`: Enables SSO for your app to help app users access your tab app seamlessly. section, which contains crucial details about your app. The application ID URI that you registered in Microsoft Entra ID is configured with the scope of the API you exposed. Configure your app's subdomain URI in `resource` to ensure that the authentication request using `getAuthToken()` is from the domain given in the app manifest. For more information, see [webApplicationInfo](../resources/schema/manifest-schema.md#webapplicationinfo).
```

Comment:
Please change tab app to API based ME app in the whole doc

Reply:
Updated.